superset-frontend/src/explore/components/controls/TimeSeriesColumnControl/TimeSeriesColumnControl.test.tsx

```diff
@@ -0,0 +1,165 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import userEvent from '@testing-library/user-event';
+import TimeSeriesColumnControl from '.';
+
+test('renders with default props', () => {
+  render(<TimeSeriesColumnControl />);
+  expect(screen.getByText('Time series columns')).toBeInTheDocument();
+  expect(screen.getByRole('button')).toBeInTheDocument();
+});
+
+test('renders popover on edit', () => {
+  render(<TimeSeriesColumnControl />);
+  userEvent.click(screen.getByRole('button'));
+  expect(screen.getByRole('tooltip')).toBeInTheDocument();
+  expect(screen.getByText('Label')).toBeInTheDocument();
+  expect(screen.getByText('Tooltip')).toBeInTheDocument();
+  expect(screen.getByText('Type')).toBeInTheDocument();
+});
+
+test('triggers onChange when type changes', () => {
+  const onChange = jest.fn();
+  render(<TimeSeriesColumnControl onChange={onChange} />);
+  userEvent.click(screen.getByRole('button'));
+  userEvent.click(screen.getByText('Select...'));
+  userEvent.click(screen.getByText('Time comparison'));
+  expect(onChange).toHaveBeenCalled();
```

Comment:
Same thing as the other tests. Any of the previous actions (any one of them) may have triggered the "onChange"

Reply:
The same observation for almost all tests

Reply:
Done.